impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Radar.js

```diff
@@ -0,0 +1,284 @@
+        }, this);
+      }, this);
+
+      return indicators;
+    },
+
+    _buildRadarRecords: function(dataTable, visualData, radarIndicators) {
+      var records = [];
+
+      // A record stands for a line of dots.
+      // One record per series value.
+      // Each record has one point per category value x measure name.
+
+      visualData.series.forEach(function(seriesEntity) {
+        var seriesTableView = this._getEntityTableView(seriesEntity, dataTable);
+        records.push(buildRecord(seriesEntity, seriesTableView));
+      }, this);
+
+      return records;
+
+      function buildRecord(seriesEntity, seriesTableView) {
+        var values = [];
+        var labels = [];
+        var record = {
+          // May be "" for the virtual null series.
+          name: seriesEntity.label,
+          value: values,
+          _label: labels
+        };
+        
+        // Given that series x category together form the
+        // [visual key]{@link pentaho.visual.role.AbstractPropertyType#isVisualKey},
+        // there can be a single data row per series and category combination.
+        // Because of this, there's no need to sum measures for a series x category combination.
+        // The value of the measure for the single data row can be used directly.
+
+        // 1st phase.
+        // Create a map/index from category key to row index.
+        var rowIndexByCategory = buildSeriesRowIndexByCategory(seriesTableView);
+
+        // 2nd phase.
+        // Now build the record values ECharts structure in the same order
+        // of the overall categories entities x measures,
+        // matching the order of indicators/axes.
+        radarIndicators.forEach(function(indicator) {
+          var category = indicator._category;
+          var measureColIndex = indicator._measure;
+          var rowIndex = O.getOwn(rowIndexByCategory, category.key);
+
+          var value = rowIndex != null
+            ? dataTable.getValue(rowIndex, measureColIndex)
+            : null;
+          var label = rowIndex != null
+              ? dataTable.getFormattedValue(rowIndex, measureColIndex)
+              : "";
+          values.push(value);
+          labels.push(label);
+        });
+
+        return record;
+      }
+
+      function buildSeriesRowIndexByCategory(seriesTableView) {
+        var rowIndexByCategory = Object.create(null);
+        var rowCount = seriesTableView.getNumberOfRows();
+        for (var rowViewIndex = 0; rowViewIndex < rowCount; rowViewIndex++) {
+          // Recover the corresponding visual row.
+          var rowIndex = seriesTableView.getSourceRowIndex(rowViewIndex);
+          var visualRow = visualData.rows[rowIndex];
+
+          rowIndexByCategory[visualRow.category.key] = rowIndex;
+        }
+
+        return rowIndexByCategory;
+      }
+    },
+
+    _configureData: function(options, radarData) {
+      options.series.forEach(function(row) {
+        row.data = radarData.records;
+      });
+    },
```

Comment:
Can be addressed on a follow-up PR.
This is still a bit "ugly", revealing that the base class may need to be changed to better accommodate for the various cases.

Reply:
Addressed. Data ( or in this case `radarData` ) is passed as an argument to `_buildSeries` https://github.com/pentaho/pentaho-platform-plugin-common-ui/pull/1760/files#diff-5ce95ff23a939baa8f5f97a800275857cc295436cc15f09262d5fa204829528aR290 .Did the same for Funnel and Gauge.